features/netsocket/NetworkInterface.h

```diff
@@ -253,6 +253,18 @@ class NetworkInterface: public DNS {
      */
     virtual nsapi_error_t gethostbyname_async_cancel(int id);
 
+    /** Get a domain name server from a list of servers to query
+     *
+     *  Returns a DNS server address for a index. If returns error no more
+     *  DNS servers to read.
+     *
+     *  @param index    Index of the DNS server, starts from zero
+     *  @param address  Destination for the host address
+     *  @param interface_name  Network interface name
+     *  @return         NSAPI_ERROR_OK on success, negative error code on failure
+     */
+    virtual nsapi_error_t get_dns_server(int index, SocketAddress *address, const char *interface_name);
```

Comment:
This is actually API break, as the size of the virtual table changes, and therefore forces us to rebuild all binary WiFi drivers that inherit NetworkInterface.

Is this new API really required?

Reply:
Hi, @tymoteuszblochmobica 
As @SeppoTakalo shared his concern on adding the new API call, can we divide this PR to add_dns_server() implementation and get_dns_server() connection separately? What customer really need urgently is add_dns_server() call.

Reply:
Member get_dns_server is not urgenly needed  as add_dns_server  so is removed.